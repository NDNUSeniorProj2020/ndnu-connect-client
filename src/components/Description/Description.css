.description-container {
	background-color: #21285C;
    color: #ffffff;
    height: 100%;
    width: 50%;
    position: absolute;
    left: 0;
    overflow: hidden;

}

.header{
    color: #ffffff;
}

h1 {
margin-left: 20px;
}

p {
   margin-left: 20px;
   margin-right: 20px;

}
