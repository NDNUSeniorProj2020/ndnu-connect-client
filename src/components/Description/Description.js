import React, { Component } from 'react';
import './Description.css';
export default class Description extends Component {
  render() {
    return (
      <view style={{flex:1, flexDirection:'row'}}>
      
      <div className="description description-container">
        { this.props.children }
        <h1 className="header">Welcome to NDNU Connect</h1>
        <div className= "info-container">
          <p>NDNU Connect is a tool for students to connect with tutors as well as NDNU's Alumni</p>
        </div>
      </div>
      </view>
    )
  }
}
