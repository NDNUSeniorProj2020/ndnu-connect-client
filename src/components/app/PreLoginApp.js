import React from 'react';

import logo from '../../assets/ndnu-logo.jpg';
import Description from '../Description/Description'
import { LoginForm, SignupForm } from '../Auth/index';

export default function PreLoginApp({ handleLogin }) {
	return (
		<div className={'App'}>
			<div className="container">
				<div className="header">
					<img src={logo} className="ndnu-logo" alt="NDNU Logo" />
					<LoginForm handleLogin={handleLogin} />
				</div>
				<div className="description">
					<Description />
				</div>
				<div className="form">
					<SignupForm />
				</div>
			</div>
		</div>
	);
}