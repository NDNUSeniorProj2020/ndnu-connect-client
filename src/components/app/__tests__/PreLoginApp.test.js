import React from 'react';
import { shallow, mount } from 'enzyme';

import PreLoginApp from '../PreLoginApp';
import Description from '../../Description/Description';

/*
* Don't worry about warning signs in our editor/IDE
* describe comes with Jest which is the default test runner
 */
describe('Testing PreLoginApp component', () => {
	it('should render without crashing.', () => {
		const tree = shallow(<PreLoginApp handleLogin={user => jest.fn(user)} />);
		expect(tree).toMatchSnapshot();
	});

	it('should contain an img tag with className ndnu-logo', () => {
		const wrapper = mount(<PreLoginApp />);
		expect(wrapper.find('img').exists()).toBe(true);
		expect(wrapper.find('img').find('.ndnu-logo').exists()).toBe(true)
	});

	it('should contain a Description component', () => {
		const wrapper = mount(<PreLoginApp />);
		expect(wrapper.find(Description).exists()).toBe(true);
	});

	it('should use default handleLogin and handleSignup props if none are passed', () => {
		expect(PreLoginApp.defaultProps.handleLogin).toBeDefined();
		expect(PreLoginApp.defaultProps.handleSignup).toBeDefined();
	});

	it('should have handleLogin and handleSignup functions as props if they are passed', () => {
		const handleLogin = user => jest.fn(user);
		const handleSignup = user => jest.fn(user);
		const wrapper = mount(<PreLoginApp handleSignup={handleSignup} handleLogin={handleLogin} />);

		expect(wrapper.props().handleLogin).toBeDefined();
		expect(wrapper.props().handleSignup).toBeDefined();
	});
});
