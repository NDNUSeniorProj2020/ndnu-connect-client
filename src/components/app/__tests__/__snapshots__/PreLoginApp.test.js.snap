// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PreLoginApp component should render without crashing. 1`] = `ShallowWrapper {}`;
